lib/howl/buffer.moon

```diff
@@ -267,6 +268,18 @@ class Buffer extends PropertyObject
       b_end_pos = @byte_offset end_pos
       @_buffer\ensure_styled_to pos: b_end_pos
 
+  mode_at: (pos) =>
+    b_pos = @byte_offset pos
+    marker = @_buffer.styling\get_nearest_style_marker b_pos
+    if marker then mode.by_name marker.name else @mode
```

Comment:
There's no guarantee that a mode with the given name exists (rather the opposite), so we need to fall to `@mode` here. Good candidate for a spec.